spec/unit/semver_spec.rb

```diff
@@ -212,6 +212,17 @@
       expect(version.tiny).to    eq(5)
       expect(version.special).to eq('-beta6')
     end
+
+    it 'should not log a deprecation warning when strict == off' do
+      Puppet[:strict] = :off
+      SemVer.new('1.2.3')
+      expect(@logs).to be_empty
+    end
+
+    it 'should log a deprecation warning unless strict == off' do
```

Comment:
This actually tests that it does not log by default (since strict ==:warning by default). It does not check the strict == :warning and strict == :error cases.